mondrian/src/main/java/mondrian/rolap/RolapSchemaPool.java

```diff
@@ -371,10 +372,43 @@ private static String getSchemaContent(
                 }
             }
         }
+        else {
+            // If exist a DynamicSchemaProcessor apply locale
+            if ( !Util.isEmpty( dynProcName ) ) {
+                return processDynamicSchemaWithCatalog( dynProcName, catalogStr, connectInfo );
+            }
+        }
 
         return catalogStr;
     }
 
+    private static String processDynamicSchemaWithCatalog(
```

Comment:
Can we refactor the code of processDynamicSchema() and make it more generic and merge its code with this new method? They largely do the same work and duplicate code.

Reply:
**Solution 1**
With the solution above, we can dismiss this method and call directly the new method:

`public String processSchema( String schema, Util.PropertyList connectInfo) throws Exception;` 

**Solution 2**
Instead call a different method, call only post-processing to apply translations.